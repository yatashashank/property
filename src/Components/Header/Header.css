.header {
    height: 90px;
    display: flex;
    border-bottom: 1px solid lightgray;
    background-color: #FAFAFB;
}

.left-part {
    display: flex;
}

.right-part {
    flex: 1;
    display: flex;
    flex-direction: row-reverse;
    margin-right: 60px;
}

.react-dropdown-h {
    font-size: 14px;
}