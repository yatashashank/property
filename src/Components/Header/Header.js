import React, { Component } from 'react';
import "./Header.css";
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from "@material-ui/icons/Menu"
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import 'react-dropdown/style.css'
import imgLogo from '../../img/logo.jpg'

class Header extends Component{
    constructor(props){
      super(props);
      this.state = {
        searchTerm: ""
    }
    }

    showMenu(){
        this.props.onclick()
    }
    render() {
        return (
            <div className="header">
                <div className="left-part">
                    <div style={{ width: 50, marginTop: 20, marginLeft: 10 }}>
                        <IconButton>
                            <MenuIcon size="medium" onClick={()=>this.showMenu()}/>
                        </IconButton></div>

                    <img src={imgLogo} alt={"Logo"} style={{ marginTop: 10, marginLeft: 10 }} width="74" height="74" />
                
                </div>
                <div className="right-part">
                    
                 <Button style={{ marginTop: 25, marginLeft: 20, height: 10 }}
                        variant="outlined"
                        color="primary"> Get Property</Button>
                <TextField
                        label="Search"
                        value={this.state.searchTerm}
                        style={{ marginLeft: 40, width: 250, marginTop: 10 }}
                    />

                </div>
         
            </div >
        );
    }
}

export default Header;
