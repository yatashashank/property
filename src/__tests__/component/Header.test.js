import React from 'react';
import toJson from 'enzyme-to-json';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow,render,mount } from 'enzyme';

import Header from '../../Components/Header/Header'
import MenuIcon from "@material-ui/icons/Menu"
import TextField from '@material-ui/core/TextField';


describe('<Header />', () => {

  it('snapshot testing for Header component', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('state value for header component',()=>{
    const wrap=mount(<Header show={true}/>);
    const val=wrap.state()
    expect(val.searchTerm).toEqual("")
})

  it('checking incoming props',()=>{
    const wrap=mount(<Header show={true}/>);
    const val=wrap.props()
   // console.log('props value ',val)
    expect(val).toBeTruthy()
  })

  it('material ui component',()=>{
    const wrap=shallow(<Header/>);
    const val=wrap.find(MenuIcon)
    expect(val.length).toBe(1)
  })

  it('onclick function simulation',()=>{
    const wrap=shallow(<Header />);
    const val=wrap.find(MenuIcon);
    //val.simulate('click').find(wrap.prop().onclick);
  
  })

  
  it('class name check',()=>{
    const wrap=shallow(<Header />);
    const val=wrap.find('div.header');
    expect(val.hasClass('header')).toBeTruthy()
  })

  it('text field value check',()=>{
    const wrap=mount(<Header />);
    const val=wrap.find(TextField);
    expect(val.length).toBe(1)
  })


})