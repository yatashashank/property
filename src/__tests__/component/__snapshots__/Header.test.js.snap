// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot testing for Header component 1`] = `
<div
  className="header"
>
  <div
    className="left-part"
  >
    <div
      style={
        Object {
          "marginLeft": 10,
          "marginTop": 20,
          "width": 50,
        }
      }
    >
      <WithStyles(IconButton)>
        <pure(MenuIcon)
          onClick={[Function]}
          size="medium"
        />
      </WithStyles(IconButton)>
    </div>
    <img
      alt="Logo"
      height="74"
      src="logo.jpg"
      style={
        Object {
          "marginLeft": 10,
          "marginTop": 10,
        }
      }
      width="74"
    />
  </div>
  <div
    className="right-part"
  >
    <WithStyles(Button)
      color="primary"
      style={
        Object {
          "height": 10,
          "marginLeft": 20,
          "marginTop": 25,
        }
      }
      variant="outlined"
    >
       Get Property
    </WithStyles(Button)>
    <TextField
      label="Search"
      required={false}
      select={false}
      style={
        Object {
          "marginLeft": 40,
          "marginTop": 10,
          "width": 250,
        }
      }
      value=""
      variant="standard"
    />
  </div>
</div>
`;
